x/auction/abci.go

```diff
@@ -13,42 +13,45 @@ import (
 func BeginBlocker(ctx sdk.Context, k keeper.Keeper, assetKeeper expected.AssetKeeper, collectorKeeper expected.CollectorKeeper, esmKeeper expected.EsmKeeper) {
 	defer telemetry.ModuleMeasureSince(types.ModuleName, ctx.BlockTime(), telemetry.MetricKeyBeginBlocker)
 
-	_ = utils.ApplyFuncIfNoError(ctx, func(ctx sdk.Context) error {
-		auctionMapData, auctionMappingFound := collectorKeeper.GetAllAuctionMappingForApp(ctx)
-		if auctionMappingFound {
-			for _, data := range auctionMapData {
-				killSwitchParams, _ := esmKeeper.GetKillSwitchData(ctx, data.AppId)
-				esmStatus, found := esmKeeper.GetESMStatus(ctx, data.AppId)
-				status := false
-				if found {
-					status = esmStatus.Status
-				}
+	auctionMapData, auctionMappingFound := collectorKeeper.GetAllAuctionMappingForApp(ctx)
+	if auctionMappingFound {
+		for _, data := range auctionMapData {
+			killSwitchParams, _ := esmKeeper.GetKillSwitchData(ctx, data.AppId)
+			esmStatus, found := esmKeeper.GetESMStatus(ctx, data.AppId)
+			status := false
+			if found {
+				status = esmStatus.Status
+			}
+			_ = utils.ApplyFuncIfNoError(ctx, func(ctx sdk.Context) error {
 				err1 := k.SurplusActivator(ctx, data, killSwitchParams, status)
 				if err1 != nil {
 					ctx.Logger().Error("error in surplus activator")
```

Comment:
return error instead of logging it.
anywhere if you use ApplyFuncIfNoError you have return error instead of logging it to make actomic state change work

Reply:
returned err after logging it now, didn't remove logger cos it would be easy to trace back to err.